src/main/java/seedu/trackascholar/logic/parser/FindCommandParser.java

```diff
@@ -16,18 +27,102 @@ public class FindCommandParser implements Parser<FindCommand> {
+            applicantPredicateList.add(nameKeywordsPredicate);
+        }
+        if (hasPrefixPresent(argMultimap, PREFIX_SCHOLARSHIP)) {
+            List<String> scholarshipKeywords = getPartialKeywordsList(argMultimap, PREFIX_SCHOLARSHIP);
+            ScholarshipContainsKeywordsPredicate scholarshipKeywordsPredicate =
+                    new ScholarshipContainsKeywordsPredicate(scholarshipKeywords);
+            applicantPredicateList.add(scholarshipKeywordsPredicate);
+        }
+        if (hasPrefixPresent(argMultimap, PREFIX_MAJOR)) {
+            List<String> majorKeywords = getPartialKeywordsList(argMultimap, PREFIX_MAJOR);
+            MajorContainsKeywordsPredicate majorKeywordsPredicate = new MajorContainsKeywordsPredicate(majorKeywords);
+            applicantPredicateList.add(majorKeywordsPredicate);
+        }
+
+        assert !applicantPredicateList.isEmpty();
+
+        Predicate<Applicant> predicate = combinePredicateList(applicantPredicateList);
+        return new FindCommand(predicate);
+    }
+
+
+    /**
+     * Combines the given predicate list into a chain of predicate.
+     *
+     * @param applicantPredicateList Predicate list to combine.
+     * @return A chain of combined predicate list.
+     */
+    private static Predicate<Applicant> combinePredicateList(List<Predicate<Applicant>> applicantPredicateList) {
+        return applicantPredicateList.stream().reduce(Predicate::or).orElse(x -> false);
+    }
+
+    /**
+     * Returns true if any of the prefixes does not contain empty {@code Optional} values in the given
+     * {@code ArgumentMultimap}.
+     */
+    private static boolean hasAnyPrefixesPresent(ArgumentMultimap argumentMultimap, Prefix... prefixes) {
+        return Stream.of(prefixes).anyMatch(prefix -> argumentMultimap.getValue(prefix).isPresent());
+    }
+
+    /**
+     * Returns true if the prefix does not contain an empty {@code Optional} value in the given
+     * {@code ArgumentMultimap}.
+     */
+    private static boolean hasPrefixPresent(ArgumentMultimap argumentMultimap, Prefix prefix) {
+        return argumentMultimap.getValue(prefix).isPresent();
+    }
+
+    /**
+     * Retrieves keyword(s) mapped by {@code Prefix} from {@code ArgumentMultimap} and splits them
+     * by whitespaces to allow for partial searching.
+     *
+     * @return List of partial keywords.
+     * @throws ParseException if the user input does not conform the expected format.
+     */
+    private static List<String> getPartialKeywordsList(ArgumentMultimap argumentMultimap,
+                                                       Prefix prefix) throws ParseException {
+        if (argumentMultimap.getValue(prefix).isEmpty()) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_USAGE));
+        }
+        List<String> keywords = argumentMultimap.getAllValues(prefix);
+        List<String> partialKeywords = new ArrayList<>();
```

Comment:
Suggestions to improve clarity for variable names: 
Rename `#getPartialKeywordsList` to `#getKeywordsLlist`
Rename `keywords` to `args` or `input`,
Rename `partialKeywords` to `keywords`

Reply:
Noted, changes made